openpype/hosts/blender/plugins/load/load_blendscene.py

```diff
@@ -0,0 +1,223 @@
+        loaded_collections.add(bpy.data.collections.get(AVALON_CONTAINERS))
+        for member in members:
+            if isinstance(member, bpy.types.Object):
+                member_parents = set(member.users_collection)
+                member_transforms[member.name] = member.matrix_basis.copy()
+            elif isinstance(member, bpy.types.Collection):
+                member_parents = {
+                    c for c in bpy.data.collections if c.user_of_id(member)}
+            else:
+                continue
+
+            member_parents = member_parents.difference(loaded_collections)
+            if member_parents:
+                collection_parents[member.name] = list(member_parents)
+
+        old_data = dict(asset_group.get(AVALON_PROPERTY))
+
+        self.exec_remove(container)
+
+        family = container["family"]
+        asset_group, members = self._process_data(libpath, group_name, family)
+
+        for member in members:
+            if member.name in collection_parents:
+                for parent in collection_parents[member.name]:
+                    if isinstance(member, bpy.types.Object):
+                        parent.objects.link(member)
+                    elif isinstance(member, bpy.types.Collection):
+                        parent.children.link(member)
+            if member.name in member_transforms and isinstance(
+                member, bpy.types.Object
+            ):
+                member.matrix_basis = member_transforms[member.name]
+
+        avalon_container = bpy.data.collections.get(AVALON_CONTAINERS)
+        avalon_container.children.link(asset_group)
+
+        # Restore the old data, but reset memebers, as they don't exist anymore
+        # This avoids a crash, because the memory addresses of those members
+        # are not valid anymore
+        old_data["members"] = []
+        asset_group[AVALON_PROPERTY] = old_data
+
+        new_data = {
+            "libpath": libpath,
+            "representation": str(representation["_id"]),
+            "parent": str(representation["parent"]),
+            "members": members,
+        }
+
+        imprint(asset_group, new_data)
+
+    def exec_remove(self, container: Dict) -> bool:
+        """
+        Remove an existing container from a Blender scene.
+        """
+        group_name = container["objectName"]
+        asset_group = bpy.data.collections.get(group_name)
+
+        attrs = [
+            attr for attr in dir(bpy.data)
+            if isinstance(
+                getattr(bpy.data, attr),
+                bpy.types.bpy_prop_collection
+            )
+        ]
+
+        members = asset_group.get(AVALON_PROPERTY).get("members", [])
+
+        for attr in attrs:
+            for data in getattr(bpy.data, attr):
+                if data in members:
+                    # Skip the asset group
+                    if data == asset_group:
+                        continue
+                    getattr(bpy.data, attr).remove(data)
```

Comment:
```suggestion
        attrs = []
        for attr_name in dir(bpy.data):
            attr = getattr(bpy.data, attr_name)
            if not isinstance(attr, bpy.types.bpy_prop_collection):
                continue
            attrs.append(attr)

        members = asset_group.get(AVALON_PROPERTY).get("members", [])

        for attr in attrs:
            for data in attr:
                if data in members:
                    # Skip the asset group
                    if data == asset_group:
                        continue
                    attr.remove(data)
```

Would this be more readable?

Also, `for data in attr` but also removing data as we iterate - doesn't that offset the indices as we iterate over it? Shouldn't we be making an explicit copy to iterate over, e.g.:

```suggestion
        attrs = []
        for attr_name in dir(bpy.data):
            attr = getattr(bpy.data, attr_name)
            if not isinstance(attr, bpy.types.bpy_prop_collection):
                continue
            attrs.append(attr)

        members = asset_group.get(AVALON_PROPERTY).get("members", [])

        for attr in attrs:
            # ensure to make a list copy because we
            # we remove members as we iterate
            for data in list(attr):
                if data in members:
                    # Skip the asset group
                    if data == asset_group:
                        continue
                    attr.remove(data)
```

Reply:
Very good point 😅 fixed it!